src/pal/src/map/virtual.cpp

```diff
@@ -1096,6 +1096,7 @@ VIRTUALCommitMemory(
         }
 
         StartBoundary = pInformation->startBoundary + runStart * VIRTUAL_PAGE_SIZE;
+        pRetVal = (void *)StartBoundary;
```

Comment:
Ah, so we had a bug in there, I have not realized that. I think your fix is correct assuming that there can only be a single run in the case the MEM_RESERVE was passed to VirtualAlloc with MEM_COMMIT or the case when we have made the reservation in the VIRTUALCommitMemory as an implicit MEM_RESERVE. Which I think is the case.